substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/ClassEntry.java

```diff
@@ -317,7 +317,7 @@ public boolean hasCompiledEntries() {
         return compiledEntryCount() != 0;
     }
 
-    public int compiledEntriesBase() {
+    public long compiledEntriesBase() {
```

Comment:
This change to the debug info model code is not appropriate. We are nowhere near being in a position where we will have DWARF sections or code (`.text`) sections whose length exceeds 32 bits. Likewise for the Microsoft CV records. So, a model for that employs `int` offsets is perfectly adequate *as a model*. The fact that a specific back end needs to write out 64 bit absolute addresses is not a valid reason to force a change like this on the model.

If the MACH-O/DWARF back end needs to write DWARF using an 8-bit offset format then we need to change the DWARF generation code so that it can be configured to support 4- or 8-bit and configure it accordingly.

If the MACH-O/DWARF back end wants abuse the DWARF model and write 8-bit absolute addresses into some of the DWARF record slots that represent offsets then it should achieve that by converting the offsets to absolute addresses in the back end, applying whatever translation it needs at the point of writing.

I'm not yet convinced we need to do that last step anyway which is an even stronger reason not to make this change.